### PR TITLE
Initialize testing framework setup

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR adds a README to the `tests` directory, as well as moving things around to start setting up for a testing framework. Discussions will follow.

### CODE REVIEW

File: echopype/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- utils
+- visualize
+- core
+
+Throughout each module, there are special file called `conftest.py`.
+This file contains all of the fixtures used within that module.
+However, the `conftest.py` within the root of the `tests` directory should contain fixtures that can be used by all tests,
+therefore, the fixtures `scope` must be `"session"`.
+
+Within echopype's software module there's a module called `testing`. Within this module contains any helper functions or classes to be used in testing.
```
Comment (lsetiawan) [inline echopype/tests/README.md:30]:
```suggestion
Within echopype there's a module called `testing`. Within this module contains any helper functions or classes to be used in testing.
```

**Comment by leewujung** [comment]
@lsetiawan: thanks for setting this up -- this looks great!

@emiliom @b-reyes and I just discussed this and we decided:
- we will not merge this PR until after the v0.6.4
- but we will get together next Wednesday to take care of _all_ the needed changes!  🚀

**Comment by leewujung** [comment]
I'll use this PR as a parking space for things to remember for our hack session:
- make sure to create temp directory such as those from `tempfile` and in various tests into the hidden echopype working directory created in #896 (this will address half of #897)

**Comment by leewujung** [comment]
I have verified that all tests passed locally. Not sure what's going on with the CI in the last few runs. Some of the failures seemed to be outside of the plotting tests. The plotting tests were the ones that more often had this type of intermittent problem.

@emiliom : I think i'll wait until this last run finished and merge, regardless of whether everything passes, so that you can PR to this `test-overhaul` branch.

**Comment by emiliom** [comment]
> Update: actually I think something got messed up by the redo PRs. Maybe you could just have the branch on your fork until I resolve this?

Alright. See https://github.com/emiliom/echopype/tree/test-convert-refactor. That branch was developed on the latest `dev` as of Wednesday or early Thursday at the latest.

**Pull request merged**